rcl/src/rcl/node.c

```diff
@@ -367,14 +366,15 @@ rcl_node_fini(rcl_node_t * node)
 }
 
 bool
-rcl_node_is_valid(const rcl_node_t * node)
+rcl_node_is_valid(const rcl_node_t * node, const rcl_allocator_t * allocator)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(node, false, rcl_get_default_allocator());
+  rcl_allocator_t alloc = allocator ? *allocator : rcl_get_default_allocator();
+  RCL_CHECK_ARGUMENT_FOR_NULL(node, false, alloc);
```

Comment:
this should check that the provided allocator is valid before using it, we're in the process of exposing a macro to do so [here](https://github.com/ros2/rcl/pull/176)

Reply:
I think allocator check should be outside this function, that means we should check allocator valid before invoke the function. Otherwise, there maybe quite a bit duplicated check in some cases.

Reply:
As a user facing function, IMO it's hard to assume that every user will provide a valid argument or write an additional line for checking it passing it to the function. That's one of the reasons we check every argument that users provide to rcl functions

Reply:
We want all arguements to all functions to be checked. Just like we require all return values to be checked. If the checks become a performance problem they can be compiled out in those use cases if we use the macros.

Reply:
Ok, that is fair. Once allocator check PR is merged, I'll update this PR.

Reply:
@jwang11 FYI https://github.com/ros2/rcl/pull/176 has been merged

Reply:
@mikaelarguedas PR update done